onmt/modules/multi_headed_attn.py

```diff
@@ -175,8 +211,26 @@ def forward(self, key: Tensor, value: Tensor,
                 query, key, value = self.linear_query(query),\
```

Comment:
a sub function for l209 to l266 could be considered

Reply:
yeah one day.